lib/Dialect/Linalg/Transforms/Tiling.cpp

```diff
@@ -248,6 +259,18 @@ mlir::linalg::tileLinalgOp(OpBuilder &b, LinalgOp op,
     auto b = ScopedContext::getBuilder();
     auto loc = ScopedContext::getLocation();
     SmallVector<Value *, 4> ivValues(ivs.begin(), ivs.end());
+
+    // If we have to apply a permutation to the tiled loop nest, we have to reorder the induction
+    // variables
+    // This permutation is the right one assuming that loopRanges have previously been permuted by (i,j,k)->(k,i,j)
+    // So this permutation should be the inversePermutation of that one: (d0,d1,d2)->(d2,d0,d1)
+    if (permutation.size() > 0) {
+      SmallVector<AffineExpr, 4> perm;
+      for (auto index : permutation)
+        perm.push_back(getAffineDimExpr(index, ScopedContext::getContext()));
+      auto invPermutationMap = inversePermutation(AffineMap::get(permutation.size(), 0, perm));
```

Comment:
Let's lift this up to the top, check `invPermutationMap` for emptiness (implicitly false) and exit gracefully with `llvm::None` please.

Reply:
Ok to lift the map building of the AffineMap to the top. But `permutation` could be legally empty to denote the identity permutation (i.e. no permutation).  So we should not exit tileLinalgOp() in that case. Then, should we then explicitly  create the identity map to always apply it  in the lambda function? Or we can guard the applyMapToValues() with an if condition?

Reply:
If the permutation is empty, you could just set the map using this: https://github.com/tensorflow/mlir/blob/master/lib/IR/AffineMap.cpp#L109

If it is not empty then use your loop based version.

Then pass that through `inversePermutation` ?

Reply:
Done!